app/Jobs/ProcessNewImage.php

```diff
@@ -206,6 +206,18 @@ protected function collectMetadata(Image $image, $path)
                     'gps_altitude' => $ref * $this->fracToFloat($exif['GPSAltitude']),
                 ]);
             }
+
+            if ($this->hasGpsImgDirInfo($exif)) {
+                $image->metadata = array_merge($image->metadata, [
+                    'yaw' => $ref * $this->fracToFloat($exif['GPSImgDirection']),
```

Comment:
The `$ref` should not be used here as it is only valid for the `GPSAltitude`. Also this would fail if `GPSAltitude` is not set because then `$ref` does not exist.